pkg/remote/resource_enumeration_error_handler.go

```diff
@@ -67,7 +97,33 @@ func HandleResourceEnumerationError(err error, alerter alerter.AlerterInterface)
 	return err
 }
 
-func handleAWSError(alerter alerter.AlerterInterface, listError *remoteerror.ResourceEnumerationError, reqerr awserr.RequestFailure) error {
+func HandleResourceDetailsFetchingError(err error, alerter alerter.AlerterInterface) error {
+	listError, ok := err.(*remoteerror.ResourceScanningError)
+	if !ok {
+		return err
+	}
+
+	rootCause := listError.RootCause()
+
+	if strings.HasPrefix(rootCause.Error(), "AccessDeniedException") {
```

Comment:
It will avoid code dupplication to do this in a single if statement IMHO

```go
if strings.HasPrefix(rootCause.Error(), "AccessDeniedException") ||
   strings.Contains(rootCause.Error(), "AccessDenied") ||
   strings.Contains(rootCause.Error(), "AuthorizationError") {
		sendDetailsFetchingAlert(aws.RemoteAWSTerraform, alerter, listError)
		return nil
	}
```

Is this expected to use `HasPrefix` for `AccessDeniedException` and `Contains` for the others ?

Reply:
> Is this expected to use HasPrefix for AccessDeniedException and Contains for the others ?

Yes, the error code isn't always the prefix of the error. Sometimes it's in the middle of the message.

Reply:
> > Is this expected to use HasPrefix for AccessDeniedException and Contains for the others ?
> 
> Yes, the error code isn't always the prefix of the error. Sometimes it's in the middle of the message.

🆗